src/v3/webpack.release.config.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { resolve } from 'path';
+import { readFileSync } from 'fs';
+
+import webpack from 'webpack';
+import { merge } from 'webpack-merge';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+import TerserPlugin from 'terser-webpack-plugin';
+import { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
+
+import terserOptions from '../../scripts/buildtools/terser/config';
+import baseConfig from './webpack.common.config';
+
+import type { Configuration } from 'webpack';
+
+const prodConfig: Configuration = merge(
+  baseConfig,
+  {
+    mode: 'production',
+    devtool: 'source-map',
+    entry: {
+      default: {
+        import: [
+          // polyfill must appear first in entry point array
+          resolve(__dirname, '../..', './polyfill/modern.js'),
+          resolve(__dirname, 'src/index.ts'),
+        ],
+        filename: 'js/okta-sign-in.min.js',
+        library: {
+          name: 'OktaSignIn',
+          type: 'umd',
+          export: 'default',
+        },
+      },
+      noPolyfill: {
+        import: [
+          resolve(__dirname, 'src/index.ts'),
+        ],
+        filename: 'js/okta-sign-in.no-polyfill.min.js',
+        library: {
+          name: 'OktaSignIn',
+          type: 'umd',
+          export: 'default',
+        },
+      },
+    },
+    resolve: {
+      alias: {
+        duo_web_sdk: 'duo_web_sdk',
+      }
+    },
+    plugins: [
+      new MiniCssExtractPlugin({
+        filename: 'css/okta-sign-in.min.css',
+      }),
+      // OKTA-429162: webpack-bundle-analyzer does not report bundled modules stats after
+      // upgrade to webpack@5
+      new BundleAnalyzerPlugin({
+        openAnalyzer: false,
+        reportFilename: 'okta-sign-in.analyzer.next.html',
+        analyzerMode: 'static',
+        defaultSizes: 'stat',
+      }),
+      new webpack.DefinePlugin({
```

Comment:
nit: move to common config then conditionally assign value based on webpack `mode`

Reply:
Partially related to this comment: https://github.com/okta/okta-signin-widget/pull/3325/files#r1283712828

I found it a lot easier to understand the configs if there aren't multiple places where the config can vary based on some variable, so duplicating this for `dev` and `release` simplifies reading it since you only need to mentally merge the two config objects rather than also needing to think about the `mode` setting to resolve what the resulting config will be.